apps/examples/src/examples/snapshot-image/SnapshotImageExample.tsx

```diff
@@ -0,0 +1,74 @@
+import { Editor, StoreSnapshot, TLPageId, TLRecord, Tldraw, TldrawImage } from '@tldraw/tldraw'
+import '@tldraw/tldraw/tldraw.css'
+import { useCallback, useState } from 'react'
+import initialSnapshot from './snapshot.json'
+
+// There's a guide at the bottom of this file!
+
+export default function SnapshotImageExample() {
+	const [editor, setEditor] = useState<Editor>()
+	const [snapshot, setSnapshot] = useState<StoreSnapshot<TLRecord>>(initialSnapshot)
+	const [pageId, setPageId] = useState<TLPageId | undefined>()
+	const [editing, setEditing] = useState(false)
+
+	const editDrawing = useCallback(() => {
+		setEditing(true)
+	}, [])
+
+	const saveDrawing = useCallback(() => {
+		if (!editor) return
+		setPageId(editor.getCurrentPageId())
+		setSnapshot(editor.store.getSnapshot())
+		setEditing(false)
+	}, [editor])
+
+	const handleMount = useCallback(
+		(editor: Editor) => {
+			setEditor(editor)
+			editor.updateInstanceState({ isDebugMode: false })
+			if (pageId) editor.setCurrentPage(pageId)
+		},
+		[pageId]
+	)
+
+	return (
+		<div style={{ padding: 30 }}>
+			<button
+				style={{ cursor: 'pointer', fontSize: 18 }}
+				onClick={editing ? saveDrawing : editDrawing}
+			>
+				{editing ? '✓ Save drawing' : '✎ Edit drawing'}
+			</button>
+			<div style={{ width: 600, height: 400, marginTop: 15 }}>
+				{editing ? (
+					<Tldraw snapshot={snapshot} onMount={handleMount} />
+				) : (
+					<TldrawImage
+						//[1]
+						snapshot={snapshot}
+						// [2]
+						pageId={pageId}
+						// [3]
+						opts={{ background: false, darkMode: false }}
+					/>
+				)}
+			</div>
+		</div>
+	)
+}
+
+/*
+This example shows how to use the `TldrawImage` component to display a snapshot
+as an image. The example also allows you to toggle between editing the snapshot
+and viewing it.
+
+[1] Pass your snapshot to the `snapshot` prop of the `TldrawImage` component.
+
+[2] You can specify which page to display by using the `pageId` prop. By
+    default, the first page is shown.
+	
+[3] You can customize the appearance of the image by passing options to the
+    `opts` prop. Try changing the `background` and `darkMode` options to see
+    their effects.
+
+ */
```

Comment:
Is this too granular? or is it ok?
cc @Taha-Hassan-Git

Reply:
Granularity is ok! Something I've been doing with these annotations is when a concept comes up that isn't directly relevant to the example, I've given a quick overview and pointed them to docs or a more relevant example so they can explore it in detail. You're using quite a cool trick to access the editor from outside of tldraw, and while it shouldn't be the focus, it's def worth mentioning imo.

You could:

- Find a way to show this concept without the extra complexity, and save this for an editor api example.
- Annotate and explain the remaining code step by step. (in which case I'd bundle these 3 comments together into one)
- Use a single annotation to give a brief overview of the remaining code.
- Have something in the introduction to explain where most of the relevant code is, and what the rest of it does.